### PR TITLE
Aggregations Refactor: Refactor Sampler Aggregation

### PR DESCRIPTION
**Pull request by colings86**

### CODE REVIEW

**Comment by colings86** [comment]
Note that this PR includes splitting the sampler aggregation into "sampler" and "diversified_sampler" aggregations for the API to make validation and parsing easier

File: core/src/main/java/org/elasticsearch/search/aggregations/bucket/sampler/DiversifiedSamplerAggregationBuilder.java [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+    }
+
+    public DiversifiedSamplerAggregationBuilder executionHint(String executionHint) {
+        this.executionHint = executionHint;
+        return this;
+    }
+
+    @Override
+    protected XContentBuilder doInternalXContent(XContentBuilder builder, Params params) throws IOException {
+        // builder.startObject();
```
Comment (jpountz) [inline core/src/main/java/org/elasticsearch/search/aggregations/bucket/sampler/DiversifiedSamplerAggregationBuilder.java]: is there value in keeping this commented out?

File: core/src/test/java/org/elasticsearch/search/aggregations/bucket/SamplerIT.java [hunk 2]
```diff
@@ -22,6 +22,7 @@
 import org.elasticsearch.action.search.SearchResponse;
 import org.elasticsearch.action.search.SearchType;
 import org.elasticsearch.index.query.TermQueryBuilder;
+import org.elasticsearch.search.aggregations.bucket.sampler.DiversifiedSamplerAggregationBuilder;
```
Comment (jpountz) [inline core/src/test/java/org/elasticsearch/search/aggregations/bucket/SamplerIT.java]: can you split it into a file for the regular sampler and another one for the diversified sampler?

**Comment by jpountz** [comment]
I actually like the split. I'm wondering that we might rename `sampler` to eg. `quality_sampler` to make it clearer how it works (as opposed to eg. a random sampler?).

**Comment by colings86** [comment]
@jpountz I pushed a commit with your suggestions implemented. I'm not a fan of `quality_sampler` as the `diversified_sampler` can also be thought of as sampling for quality purposes. Personally I think 'sampler' is fine as its the simple no frills version of sampling

**Comment by jpountz** [comment]
LGTM

**Pull request merged**